src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs

```diff
@@ -241,6 +242,9 @@ public static partial class MicrosoftIdentityWebApiAuthenticationBuilderExtensio
                             throw new UnauthorizedAccessException(IDWebErrorMessage.NeitherScopeOrRolesClaimFoundInToken);
                         }
 
+                        // Validates the token authorization rules.
+                        TokenAuthorizationRules.ValidateApiAuthorization(mergedOptions, context!.Principal!);
```

Comment:
This is all the code that would need to be added to an OnTokenValidated event in an ASP.NET Core web api that wouldn't use Ms.Id.Web, or in an ASP.NET classic app. The first member is a TokenAuthorizationRules (So 4 properties), and the second a ClaimsPrincipal, which is always available after token validation)